compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -138,41 +138,28 @@ public STNode parse(ParserRuleContext context) {
      */
     @Override
     public STNode resumeParsing(ParserRuleContext context, Object... args) {
-        // TODO: revisit the commented resume-points
+        return resumeTopLevelNodeParsing(context, args);
```

Comment:
Can't we have the logic inside this function iteself?
There's no value addition of delegating the logic to another function, if this is the only place it is used..

Reply:
yea sounds good to me